### ISSUE TITLE
There should be a block selection mode when pressing Ctrl+V

### ISSUE THREAD

**Issue by maelp** [sublimehq/Vintage#74]
The block selection mode of Vim is very practical to make multiple-line insertions or modifications :)

**Comment by misfo** [comment]
In my experience Vim's block selection is useful in a subset of the case for which Sublime's multiple selection is useful (i.e. multiple selection is more powerful).  So I'm more than happy having multiple selection instead.  Thoughts?

**Comment by maelp** [comment]
I did not know at that time that there existed a multiple selection feature,
however I'd be interested to know how to easily write a plugin such
that ctrl+v enters column multiple selection mode,
and that pressing up and down selects the column on the line above or
below, and pressing left and right is equivalent to pressing
shift+left and shift+right in multiple selection in that it does a
block selection
is that complex to write, would you mind giving me a hint as to where
I should look to understand how to do this (I know Python but I never
wrote plugins for Sublime Text, plus I wonder whether one should add
the key shortcut to the vintage shortcuts or to the Sublime Text
general key shortcuts? )

On Fri, Mar 9, 2012 at 06:38, Trent Ogren
reply@reply.github.com
wrote:

**Comment by guillermooo** [comment]
@misfo, I don't think ST has a true block selection? Vertical selections that are smaller than whole lines will necessarily be multiple selections. I'm fine with this.

So:
- CTRL-V
- 2j

...would yield three selections starting at the same column. Further horizontal movement should then work as in multiple selections. Further vertical movement should add/subtract selections.

Is that what you mean?

**Comment by misfo** [comment]
I think that _could_ work. I'm just not sure it would provide much advantage over just using the existing multiple selection functionality with Vintage mode

**Comment by maelp** [comment]
That would be great

On Fri, Mar 9, 2012 at 14:06, Trent Ogren
reply@reply.github.com
wrote:

**Comment by guillermooo** [comment]
Meaning _Ctrl+D_ and _Ctrl+U_ only? I would certainly like to be able to extend/shrink the selection vertically with _j/k_ instead of _Ctrl+Alt+Arrow Down_. TBH, I've tried to implement this already, but I gave up after a couple of attempts.

**Comment by MitchellMcKenna** [comment]
+1 for this, as a VIM user Visual Block mode is engrained into my head, not having it is a deal breaker for me

**Comment by phzbox** [comment]
I agree that Ctrl+V should do a block selection rather than paste in vim-mode (where people paste with p and P). 

@MitchellMcKenna : An easy workaround is to use "wv" instead of ctrl+v, which select the next word. More often than not, I was doing ctrl+v ww or something similar, so by _not_ having the block selection, I make my selection more accurate the first time : )

**Comment by yinrong** [comment]
+1 want ctrl+v to be vertical selction

**Comment by ethan-cao** [comment]
vote for it!

**Comment by xsong16** [comment]
+1 for this too

**Comment by slonia** [comment]
+1

**Comment by Yakubovich** [comment]
+1 (this is currently my only reason for not switching to Sublime)

**Comment by guillermooo** [comment]
@Yakubovich Have you tried Vintageous (on GH too)?

**Comment by Yakubovich** [comment]
@guillermooo thanks, that works for me!

**Comment by tonnyone** [comment]
+1 i need the Ctrl + v

**Comment by ebylund** [comment]
This is my biggest hold back for not using sublime as my full time editor. I love the out of the box vintage mode, but the block selection is a huge feature i use that i wish it had

**Comment by arasmussen** [comment]
+1 want

**Comment by jwmortensen** [comment]
+1

**Comment by yinrong** [comment]
stop using sublime. back to using VIM and find I just don't need sublime

**Comment by hanjukim** [comment]
+1 ctrl+v to column selection!!

**Comment by hkinks** [comment]
+1

**Comment by caiotomazelli** [comment]
+1

**Comment by tl3shi** [comment]
+1

**Comment by raghavrv** [comment]
+100

**Comment by trobert2** [comment]
+1

**Comment by hlzz** [comment]
+10086

**Comment by Qix-** [comment]
4 years later, still not here. Still a dealbreaker. Even if `^V` meant using Sublime's multiple selection thing, then great. The fact it isn't there in any capacity forces me not to use Sublime unfortunately.

**Comment by tomleo** [comment]
In some vim GUIs ctrl-q is used instead of ctrl-v. By using ctrl-q, you don't change the default key mapping for ctrl-v.

**Comment by ghost** [comment]
+10086

**Comment by theSha1chemist** [comment]
+ 1

**Comment by theSha1chemist** [comment]
+ 1

**Comment by theSha1chemist** [comment]
+ 1

**Comment by theSha1chemist** [comment]
\+ 1

**Comment by rschalch** [comment]
+10000

**Comment by omarps** [comment]
+1

**Comment by happyHou** [comment]
+1

**Comment by fut33v** [comment]
+1

**Comment by yudai524** [comment]
+1

**Comment by happyHou** [comment]
+1

**Comment by mikeybkats** [comment]
+1

**Comment by mgresko** [comment]
+1

**Comment by mgresko** [comment]
+1

**Comment by marciol** [comment]
+1

**Comment by odjhey** [comment]
+1111 any update on this?

**Comment by Qix-** [comment]
# This hasn't been updated in almost 5 years. This won't be fixed. They have abandoned this project. Find another Vim plugin. '+1'-ing it won't help.

**Comment by roskow12** [comment]
+1

**Comment by houming818** [comment]
+1 want ctrl+v to be vertical selction

**Comment by tgudlek** [comment]
+1

**Comment by pyq0109** [comment]
+1s

**Comment by lvvorz** [comment]
+10086s

**Comment by avegancafe** [comment]
+1

**Comment by mfournial** [comment]
+1

**Comment by ksttest** [comment]
+1

**Comment by samwestmoreland** [comment]
+1

**Comment by avegancafe** [comment]
Is there any reason for this not to be updated? This is a built-in plugin in sublime. If it's abandoned, it should probably be taken out

**Comment by anaskhan96** [comment]
It's real sad to see they've abandoned this issue as it is. Visual Block mode is a critical feature, and there should've been atleast some steps taken toward addressing its absence, specially if there's support from users in the community over a course of almost 6 years now.

**Comment by avegancafe** [comment]
++ @anaskhan96 . This is the sole reason that I refuse to use sublime. I use like 7 different editors and sublime beats all of them in many many ways, but this is far too crucial of a feature to be missing.

**Comment by anaskhan96** [comment]
@kyleholzinger you might be interested in checking out [Vintageous](https://github.com/guillermooo/Vintageous). I just started using it as an alternative to Vintage, and for starters, they have Visual Block mode.

**Comment by avegancafe** [comment]
This is awesome @anaskhan96, I actually found [NeoVintageous](https://github.com/NeoVintageous/NeoVintageous) works better for me, but thanks for the recommendation!

**Comment by ghking1** [comment]
+1

**Comment by wsunid** [comment]
+1

**Comment by danliustc** [comment]
> This is awesome @anaskhan96, I actually found [NeoVintageous](https://github.com/NeoVintageous/NeoVintageous) works better for me, but thanks for the recommendation!

That's really awesome! I really recommend vim user like me install and config it.

**Comment by franciscolopezsancho** [comment]
Yess. Thanks!!. NeoVintageous has block-selection mode. You need to activate it though; go to Preferences -> Settings and set `"vintageous_use_ctrl_keys": true,`

**Comment by jkryanchou** [comment]
> vintageous_use_ctrl_keys

Wow! it works, I'm looking forward the solution for a long time. Excellent Job!
